application/src/Api/Adapter/SiteAdapter.php

```diff
@@ -89,6 +89,15 @@ public function hydrate(Request $request, EntityInterface $entity,
             }
             $entity->setNavigation($request->getValue('o:navigation', $default));
         }
+        if ($this->shouldHydrate($request, 'o:homepage')) {
+            $homepage = $request->getValue('o:homepage');
+            if (isset($homepage['o:id']) && is_numeric($homepage['o:id'])) {
+                $homepage = $this->getAdapter('site_pages')->findEntity($homepage['o:id']);
+            } else {
+                $homepage = null;
+            }
+            $entity->setHomepage($homepage);
+        }
```

Comment:
Should have a sanity check here or in validate that the homepage is actually one of the site's pages.